### ISSUE TITLE
Views deployment failure: Permission denied: '.probe_cache'

### ISSUE THREAD

**Issue by whd** [mozilla/bigquery-etl#2214]
/CC @acmiyaguchi 

```
+ ./script/glam/generate_glean_sql
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/app/bigquery_etl/glam/generate.py", line 254, in <module>
    main()
  File "/app/bigquery_etl/glam/generate.py", line 219, in main
    **models.histogram_bucket_counts(
  File "/app/bigquery_etl/glam/models.py", line 118, in histogram_bucket_counts
    custom_distribution_metadata_list=get_custom_distribution_metadata("fenix"),
  File "/app/bigquery_etl/glam/utils.py", line 70, in get_custom_distribution_metadata
    probes = glean.get_probes()
  File "/usr/local/lib/python3.8/site-packages/mozilla_schema_generator/glean_ping.py", line 88, in get_probes
    data = self._get_json(self.probes_url)
  File "/usr/local/lib/python3.8/site-packages/mozilla_schema_generator/generic_ping.py", line 219, in _get_json
    return json.loads(GenericPing._get_json_str(url))
  File "/usr/local/lib/python3.8/site-packages/mozilla_schema_generator/generic_ping.py", line 208, in _get_json_str
    GenericPing._add_to_cache(no_param_url, final_json)
  File "/usr/local/lib/python3.8/site-packages/mozilla_schema_generator/generic_ping.py", line 180, in _add_to_cache
    GenericPing.cache_dir.mkdir(parents=True)
  File "/usr/local/lib/python3.8/pathlib.py", line 1288, in mkdir
    self._accessor.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '.probe_cache'
```

https://github.com/mozilla/mozilla-schema-generator/blob/main/mozilla_schema_generator/generic_ping.py#L26

Same problem as #2212 , but via the MSG library looks like.

**Comment by whd** [comment]
I'm not sure why this would start happening again, but it is.

**Comment by fbertsch** [comment]
cc @wlach, it's probably happening because of the new code using MSG.

**Comment by wlach** [comment]
> cc @wlach, it's probably happening because of the new code using MSG.

Hmm. @whd can you post a newer traceback? FWIW the new code is here: https://github.com/mozilla/bigquery-etl/pull/2271

@fbertsch Do we need to create `.probe_cache` in the current working directory? `$HOME` seems like a more logical choice to me? FWIW you added this about 16 months ago in https://github.com/mozilla/mozilla-schema-generator/commit/b54ede62df10f1feab153665babf9e9b9e8ea018

**Comment by fbertsch** [comment]
@wlach `$HOME` would probably work well!

**Comment by whd** [comment]
```
+ cd /app
+ cp -R sql /tmp/sql
+ ./script/generate_all_views --target-project moz-fx-data-shared-prod --sql-dir /tmp/sql
+ python -m bigquery_etl.view.generate_stable_views --target-project moz-fx-data-shared-prod --sql-dir /tmp/sql --parallelism=20
telemetry_derived/foo/query.sql                             OK
Cannot get Bigquery schema for mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/glean/glean/glean.1.schema.json: "filename 'mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/glean/glean/glean.1.bq' not found"
Cannot get Bigquery schema for mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/credentials/credentials.1.schema.json: "filename 'mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/credentials/credentials.1.bq' not found"
Cannot get Bigquery schema for mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/metaschema/metaschema.1.schema.json: "filename 'mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/metaschema/metaschema.1.bq' not found"
Cannot get Bigquery schema for mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/pioneer-ingestion/pioneer-ingestion.1.schema.json: "filename 'mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/pioneer-ingestion/pioneer-ingestion.1.bq' not found"
Cannot get Bigquery schema for mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/sources/sources.1.schema.json: "filename 'mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/sources/sources.1.bq' not found"
Cannot get Bigquery schema for mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/structured-ingestion/structured-ingestion.1.schema.json: "filename 'mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/structured-ingestion/structured-ingestion.1.bq' not found"
Cannot get Bigquery schema for mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/telemetry-ingestion/telemetry-ingestion.1.schema.json: "filename 'mozilla-pipeline-schemas-9e81d822064c5a8336d3f2e61cc67cdf58d4a010/schemas/metadata/telemetry-ingestion/telemetry-ingestion.1.bq' not found"
+ ./script/bqetl glean_usage generate --project-id moz-fx-data-shared-prod --output-dir /tmp/sql
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/app/bigquery_etl/cli/__init__.py", line 56, in cli
    group(prog_name=prog_name)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/app/bigquery_etl/cli/glean_usage.py", line 112, in generate
    pool.map(
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 364, in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 771, in get
    raise self._value
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/lib/python3.8/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/app/bigquery_etl/glean_usage/glean_app_ping_views.py", line 57, in generate_per_app
    (r for r in GleanPing.get_repos() if r["name"] == release_app["v1_name"])
  File "/usr/local/lib/python3.8/site-packages/mozilla_schema_generator/glean_ping.py", line 207, in get_repos
    repos = GleanPing._get_json(GleanPing.repos_url)
  File "/usr/local/lib/python3.8/site-packages/mozilla_schema_generator/generic_ping.py", line 219, in _get_json
    return json.loads(GenericPing._get_json_str(url))
  File "/usr/local/lib/python3.8/site-packages/mozilla_schema_generator/generic_ping.py", line 208, in _get_json_str
    GenericPing._add_to_cache(no_param_url, final_json)
  File "/usr/local/lib/python3.8/site-packages/mozilla_schema_generator/generic_ping.py", line 180, in _add_to_cache
    GenericPing.cache_dir.mkdir(parents=True)
  File "/usr/local/lib/python3.8/pathlib.py", line 1288, in mkdir
    self._accessor.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '.probe_cache'
telemetry_derived/foo/query.sql      
```

**Comment by wlach** [comment]
@whd I just released mozilla-schema-generator 0.4.1 to pypi: https://pypi.org/project/mozilla-schema-generator/0.4.1/

I believe upgrading the dependency + setting the environment variable `MSG_PROBE_CACHE_DIR` to something like `$HOME/.probe_cache` inside the docker container should fix this issue.

**Comment by whd** [comment]
I filed https://github.com/mozilla-services/cloudops-infra/pull/3316 for setting this variable so once the dependency is updated in this repo deploys should start working again (though #2287 may prevent a views deploy from triggering).
